### PR TITLE
refactor: dependencies types

### PR DESCRIPTION
**Pull request by leogr**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

/kind design

> /kind documentation

> /kind failing-test

/kind feature

> /kind flaky-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area library

/area cli

> /area tests

> /area examples

**What this PR does / why we need it**:

This allows full support for https://github.com/falcosecurity/falco/pull/2190

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Not yet sure if `|` is a good symbol to express the "OR" meaning. Apparently yes, but one can be in trouble when using it in a shell command. Anyway, we can change it later.

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm

File: pkg/oci/types.go [hunk 1]
```diff
@@ -61,20 +65,79 @@ type RegistryResult struct {
...
+func (rc *ArtifactConfig) ParseDependencies(dependencies ...string) error {
+	for _, d := range dependencies {
+		artifactRefs := strings.Split(d, "|")
+		var insertPos int
+		for i, a := range artifactRefs {
+			parsedRef, err := artifact.ParseRef(a)
+			if err != nil {
+				return fmt.Errorf(`cannot parse "%s": %w`, a, err)
+			}
+			if i == 0 {
```
Comment (alacuku) [inline pkg/oci/types.go:135]: Some comments to explain the logic could be very useful here.

**Review by alacuku — COMMENTED** [review]
Nice PR! Just left some minor comments.

/LGTM

**Comment by leogr** [comment]
Hey @alacuku 

Thank you for your suggestions, however, it's too late for this PR :smiling_face_with_tear: 

Would you mind opening a new one?

**Pull request merged**